frontend/pages/DetailedViewPage.js

```diff
@@ -26,11 +25,35 @@ export default class DetailedViewPage extends Component {
         this.setState(({priceOverlayVisible}) => ({
             priceOverlayVisible: !priceOverlayVisible
         }));
-    }
+    };
 
-    changePrice = () => {
+    changePrice = async () => {
+        /*
+            Find the first store that matches the store selected by the user to
+            get the location data from it.
+         */
 
-    }
+        let storeName = this.props.navigation.state.params.store;
```

Comment:
I understand that it might be very hard to change, but I feel that the whole `this.props.navigation.state.params.store` is too complex and too deep from the outer most layer (this.props). I am confused by that this variable is about by just looking at the name.

Reply:
Good point, we can rename this parameter to be more clear. It is supposed to represent the name of the store the user is currently looking at.